test/test_linalg.py

```diff
@@ -2637,6 +2637,27 @@ def test_cholesky_solve_out_errors_and_warnings(self, device, dtype):
             self.assertEqual(len(w), 1)
             self.assertTrue("An output with one or more elements was resized" in str(w[-1].message))
 
+    @skipCUDAIfNoMagma
+    @skipCPUIfNoLapack
+    @dtypes(torch.float, torch.double)
+    def test_cholesky_solve_backward(self, device, dtype):
+        from torch.testing._internal.common_utils import random_hermitian_pd_matrix
+
+        # A different code path runs depending on if we ask for the grad of A.
+        for requires_A_grad in (False, True):
+            b_dims = (5, 2)
+            A_dims = (5,)
+            upper = False
+
+            b = torch.randn(*b_dims, dtype=dtype, device=device, requires_grad=True)
+            A = random_hermitian_pd_matrix(*A_dims, dtype=dtype, device=device)
+            A.requires_grad = requires_A_grad
+            L = torch.linalg.cholesky(A, upper=upper)
+            (torch.cholesky_solve(b, L, upper=upper).sum()).backward()
+
+            expected_grad = torch.linalg.solve(A, torch.ones_like(b))
+            self.assertEqual(b.grad, expected_grad)
```

Comment:
Rather than testing this way, which assumes which are the formulas for the gradinet, could you use [gradcheck](https://pytorch.org/docs/stable/generated/torch.autograd.gradcheck.html)?

Reply:
Updated the test to use `gradcheck`. I wasn't able to find a clean way to use gradcheck for the matrix part though. `cholesky_solve` is a bit weird because it ignores the lower or upper triangle of the matrix and computes the gradient as if the matrix is symmetric. When finite differencing, you just see zeros in the components that are ignored.

Reply:
left a comment below